site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,168 @@
+Retention policy can be configured both globally and on snapshot reference to provide highly flexible customization to the expiration behavior.
+
+#### Global snapshot retention policy
+
+Global snapshot retention policy can be set through the following table properties:
+
+| Property                             | Default            | Description                                                   |
+| ------------------------------------ | ------------------ | ------------------------------------------------------------- |
+| history.expire.max-snapshot-age-ms   | 432000000 (5 days) | Default max age of snapshots to keep while expiring snapshots    |
+| history.expire.min-snapshots-to-keep | 1                  | Default min number of snapshots to keep while expiring snapshots |
+
+#### Snapshot reference retention policy
+
+Similarly, snapshot reference has the properties below to provide finer grain control:
+
+| Property                     | Type      | Description |
+|------------------------------|-----------|-------------|
+| **`min-snapshots-to-keep`**  | `int`     | For `branch` type only, the minimum number of snapshots to keep in a branch, default to the current value of table property `history.expire.min-snapshots-to-keep` when this value is evaluated |
+| **`max-snapshot-age-ms`**    | `long`    | The duration before a snapshot tagged or in a branch could be expired by any automatic snapshot expiration process, default to the current value of table property `history.expire.max-snapshot-age-ms` when this value is evaluated |
+
+#### Policy evaluation mechanism
+
+When a snapshot expiration process starts, it follows the steps described below:
+
+1. form an expiration candidate pool containing all snapshots
+2. for each snapshot reference, evaluate the associated policy and move snapshots out of the candidate pool
+3. apply global retention policy to and move snapshots out of the candidate pool
+4. when multiple snapshots can be chosen to be moved out, newer snapshots win
+4. after evaluation, expire all snapshots that are still in the candidate pool
+
+#### Policy evaluation example
+
+Here is an example for how an Iceberg snapshot expiration procedure evaluates what snapshots to expire.
+
+Suppose we have the following snapshot graph and retention policies configured:
+
+```
+A -> B -> C (main)
+ \      (dev)
+  D -> E (b1)
+   \
+    F -> G (b2)
+     \
+      H (b3)
+```
+
+| Policy Type      | Max Age       | Min to Keep | Snapshots Affected       |
+|------------------|---------------|-------------|--------------------------|
+| global           | 5 hours       | 4           | A, B, C, D, E, F, G, H   |
+| branch/main      | 3 days        | 1           | A, B, C                  |
+| branch/b1        | 2 days        | 2           | A, D, E                  |
+| branch/b2        | 1 day         | 0           | A, D, F, G               |
+| tag/dev          | forever       | N/A         | C                        |
```

Comment:
How do you set "forever"? I think this would make sense if `max-ref-age-ms` is unset.